### PR TITLE
Use MMAIntrinsicAttr, not MMAAttr, when just a plain intrinsic is meant.

### PR DESCRIPTION
**Pull request by bjacob**
MMAIntrinsicAttr is the enum attribute directly wrapping the MMAIntrinsic enum.
MMAAttr is a bigger machinery implementing some codegen interfaces.

MMAAttr has a custom assembly format that makes it look almost like just the underlying MMAIntrinsicAttr in the IR: `<MFMA_XYZ>` vs `MFMA_XYZ`. Still, when we have have many of these, such as in the `mma` parameter of `IREE::GPU::TargetWgpAttr`, it's nice to drop all those `<...>`.

Writing this PR seems to have shown why authors were drawn to MMAAttr: it offers some useful methods. The new code added in this PR is mostly adding the equivalent functions, as global functions not methods, for MMAIntrinsicAttr. The code was already there, just not exposed for MMAIntrinsicAttr.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/GPU/test/gpu_materialize_encoding_gfx942.mlir [hunk 1]
```diff
@@ -640,7 +640,7 @@ func.func @matmul_lowering_MFMA_I32_16x16x32_I8() {
       storage =  b64|b32|b16|b8,
       subgroup = shuffle|arithmetic,
       dot =  dp4xi8toi32,
-      mma = [<MFMA_F32_16x16x4_F32>, <MFMA_F32_16x16x16_F16>, <MFMA_F32_32x32x8_F16>, <MFMA_F32_16x16x32_F8E4M3FNUZ>, <MFMA_I32_16x16x32_I8>, <MFMA_I32_32x32x16_I8>],
+      mma = [ MFMA_F32_16x16x4_F32, MFMA_F32_16x16x16_F16, MFMA_F32_32x32x8_F16, MFMA_F32_16x16x32_F8E4M3FNUZ, MFMA_I32_16x16x32_I8, MFMA_I32_32x32x16_I8],
```
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/Common/GPU/test/gpu_materialize_encoding_gfx942.mlir:643]: Could you clean up these leading spaces?

**Review by kuhar — COMMENTED** [review]
I think we also need to update the bindings in https://github.com/iree-org/iree/blob/e61707092020fbbf8272fe7f514dcae01c6519ee/compiler/bindings/python/IREECompilerDialectsModule.cpp#L295-L337 -- we would prefer to query the mnk/abc/shape from the intrinsic attr directly

**Review by qedawkins — COMMENTED** [review]
I was hoping we'd go the opposite direction and make it a list of `MMAInterfaceAttr`.

**Comment by bjacob** [comment]
Discussed offline. The idea is portability to other targets where the equivalent capabilities are not a good fit for enums, such as Vulkan cooperative matrix. Some kind of interface will provide the necessary polymorphism. The right interface for this doesn't exist yet, but that's the idea why we are currently using MMAAttr.

**Pull request closed without merging**